File: Lesson1.cpp
#include <iostream>

/*
 * Basic objects in C++
 */

class Lesson1 {
public:
    int x;
    static void DoSomethingElse()
    {
        std::cout << "DoSomethingElse" << std::endl;
    }
    void DoSomething(int value);
};

void Lesson1::DoSomething(int value)
{
    x = value;
    std::cout << "x is equal to " << x << "\n";
    DoSomethingElse();
}

int main()
{
    Lesson1 *obj = new Lesson1();
    obj->DoSomething(4);
    delete obj;
}

File: Lesson1.py
#!/usr/bin/env python

from __future__ import print_function

class Lesson1(object):
    def __init__(self, x = 0):
        self.x = x
    @staticmethod
    def DoSomethingElse():
        print('DoSomethingElse')
    @classmethod
    def DoSomething(self, value):
        self.x = value
        print('x is equal to {}'.format(self.x))
        Lesson1.DoSomethingElse()

obj = Lesson1()
obj.DoSomething(4)

